server/src/main/java/org/candlepin/audit/EventSource.java

```diff
@@ -67,39 +60,15 @@ protected ClientSessionFactory createSessionFactory() throws Exception {
     }
 
     protected void shutDown() {
-        try {
-            session.stop();
-            session.close();
-            factory.close();
-        }
-        catch (ActiveMQException e) {
-            log.warn("Exception while trying to shutdown ActiveMQ", e);
-        }
+        closeEventReceivers();
+        factory.close();
     }
 
-    void registerListener(EventListener listener) {
-        String queueName = QUEUE_ADDRESS + "." + listener.getClass().getCanonicalName();
-        log.debug("registering listener for {}", queueName);
-
-        try {
-            try {
-                // Create a durable queue that will be persisted to disk:
-                session.createQueue(QUEUE_ADDRESS, queueName, true);
-                log.debug("created new event queue " + queueName);
-            }
-            catch (ActiveMQException e) {
-                // if the queue exists already we already created it in a previous run,
-                // so that's fine.
-                if (e.getType() != ActiveMQExceptionType.QUEUE_EXISTS) {
-                    throw e;
-                }
-            }
+    void registerListener(EventListener listener) throws Exception {
+        this.eventReceivers.add(new EventReceiver(listener, factory, mapper));
+    }
 
-            ClientConsumer consumer = session.createConsumer(queueName);
-            consumer.setMessageHandler(new ListenerWrapper(listener, mapper));
-        }
-        catch (ActiveMQException e) {
-            log.error("Unable to register listener :" + listener, e);
-        }
+    private void closeEventReceivers() {
+        this.eventReceivers.forEach((receiver) -> { receiver.close(); });
```

Comment:
We can strip this down quite a bit to just

```java
this.eventReceivers.forEach(EventReceiver::close);
```

Reply:
:+1: